test/parallel/test-crypto.js

```diff
@@ -167,42 +167,46 @@ testImmutability(crypto.getCurves);
 
 // Regression tests for https://github.com/nodejs/node-v0.x-archive/pull/5725:
 // hex input that's not a power of two should throw, not assert in C++ land.
-assert.throws(function() {
-  crypto.createCipher('aes192', 'test').update('0', 'hex');
-}, (err) => {
-  const errorMessage =
-    common.hasFipsCrypto ? /not supported in FIPS mode/ : /Bad input string/;
-  // Throws general Error, so there is no opensslErrorStack property.
-  if ((err instanceof Error) &&
-      errorMessage.test(err) &&
-      err.opensslErrorStack === undefined) {
-    return true;
+
+common.expectsError(
+  () => crypto.createCipher('aes192', 'test').update('0', 'hex'),
+  {
+    code: common.hasFipsCrypto ? undefined : 'ERR_INVALID_ARG_VALUE',
+    type: common.hasFipsCrypto ? Error : TypeError,
+    message: common.hasFipsCrypto ? /not supported in FIPS mode/ : 'Invalid encoding for data',
+    opensslErrorStack: undefined
   }
-});
+);
 
-assert.throws(function() {
-  crypto.createDecipher('aes192', 'test').update('0', 'hex');
-}, (err) => {
-  const errorMessage =
-    common.hasFipsCrypto ? /not supported in FIPS mode/ : /Bad input string/;
-  // Throws general Error, so there is no opensslErrorStack property.
-  if ((err instanceof Error) &&
-      errorMessage.test(err) &&
-      err.opensslErrorStack === undefined) {
-    return true;
+common.expectsError(
+  () => crypto.createDecipher('aes192', 'test').update('0', 'hex'),
+  {
+    code: common.hasFipsCrypto ? undefined : 'ERR_INVALID_ARG_VALUE',
+    type: common.hasFipsCrypto ? Error : TypeError,
+    message: common.hasFipsCrypto ? /not supported in FIPS mode/ : 'Invalid encoding for data',
+    opensslErrorStack: undefined
```

Comment:
I think it makes more sense to create two different objects here:

```js
common.hasFipsCrypto ? {
  type: Error,
  message: /not supported in FIPS mode/
} : {
  code: 'ERR_INVALID_ARG_VALUE',
  message: '...'
}
```

Reply:
Fixed.